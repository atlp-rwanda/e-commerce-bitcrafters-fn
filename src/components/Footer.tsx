import React from "react";

const Footer: React.FC = () => {
  return (
    <footer>
      <div className="container">
        <p>
          &copy; {new Date().getFullYear()} BitCrafters E-commerce App. All
          rights reserved.
        </p>
      </div>
    </footer>
  );
};

export default Footer;
