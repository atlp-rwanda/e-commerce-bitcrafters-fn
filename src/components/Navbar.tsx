import React, { useState } from "react";
import Logo from "../assets/images/Bit.Shop.svg";
import { RxHamburgerMenu, RxCross2, RxAvatar } from "react-icons/rx";
import { Link } from "react-router-dom";
import { CiSearch, CiHeart, CiSettings } from "react-icons/ci";
import { IoCartOutline, IoNotificationsOutline, IoLocationOutline, IoChatbubblesOutline} from "react-icons/io5";
import { shallowEqual, useSelector } from "react-redux";
import { RootState } from "../redux/store";
import SearchComponent from "./SearchComponent";
import Badge from "@mui/material/Badge";
import "react-toastify/dist/ReactToastify.css";
import { useNotifications } from "./notificationtoast";
import NotificationPane from "../views/NotificationPane";
import { FaTimes } from "react-icons/fa";
import Chat from "../views/chat/Chat";
import { TbDashboard } from "react-icons/tb";
import Logout from "./Logout";
import { IoMdHome } from "react-icons/io";

interface NavbarProps {
  burgerShown?: boolean;
  showSearch?: boolean;
}

const Navbar: React.FC<NavbarProps> = (props) => {
  const isLoggedIn = useSelector(
    (state: RootState) => state.auth.isLoggedIn,
    shallowEqual,
  );

  const [isChatOpen, setIsChatOpen] = useState(false);
  // const dispatch = useDispatch();
  const authRole = useSelector(
    (state: RootState) => state.auth.authRole,
    shallowEqual,
  );
  const [isProfileModalOpen, setIsProfileModalOpen] = useState(false);

  const toggleChat = () => {
    setIsChatOpen(!isChatOpen);
  };
  const { count } = useSelector((state: any) => state.cart);
  const wishListCount = useSelector((state: any) => state.wishList?.count ?? 0);
  const [burgerShown, setIsBurgerShown] = useState(props.burgerShown || false);
  const [showSearch, setShowSearch] = useState(props.showSearch || false);
  const { unreadCount } = useNotifications();
  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);

  return (
    <div className="w-full flex flex-col">
      <nav className="navbar-container w-full border-b-1 p-8 flex-between space-x-2 border-b-[1px] border-gray_100">
        <div className="nav-logo w-[20%]">
          <img src={Logo} alt="Logo" className="w-24 object-contain" />
        </div>
        <div className="nav-link flex-between w-[30%] hidden tablet:flex">
          <ul className="flex-between space-x-4">
            <li data-testid="mobile-menu-home">
              <Link to="/" className="text-sm">Home</Link>
            </li>
            <li>
              <Link to="/about" className="text-sm">About</Link>
            </li>
            <li>
              <Link to="/" className="text-sm">Shop</Link>
            </li>
            <li>
              <Link to="/" className="text-sm">Contact</Link>
            </li>
          </ul>
        </div>
        <div className="nav-icons flex-between hidden tablet:flex">
          <ul className="flex-between space-x-4">
            <li>
              <div className="search-container bg-gray rounded-sm flex-between space-x-2 p-2 cursor-pointer" onClick={() => setShowSearch(!showSearch)}>
                <p className="text-xs">search here</p>
                <CiSearch />
              </div>
            </li>
            {isLoggedIn && (
              <>
                <li>
                  <Link to="/wishList" className="text-lg">
                    {" "}
                    <Badge badgeContent={wishListCount} color="primary">
                      <CiHeart size={24} aria-label="Heart" />
                    </Badge>
                  </Link>
                </li>
                <li>
                  <Link to="/cart" className="text-lg relative">
                    <IoCartOutline size={24} aria-label="Cart" />
                    {count > 0 && (
                      <p className="bg-red-500 p-1 text-1 flex items-center justify-center text-white absolute rounded-full bottom-[-3px] right-[-3px] text-[8px] h-[13px] w-[13px]">{count}</p>
                    )}
                  </Link>
                </li>
                <li>
                  <Link to="" className="text-lg" data-testid="notifications-link">
                    <Badge badgeContent={unreadCount} color="primary">
                      <IoNotificationsOutline size={24} onClick={() => setIsModalOpen(true)} />
                    </Badge>
                  </Link>
                </li>
                <li>
                  <Link to="/orders" className="text-lg" data-testid="orders-link">
                    <IoLocationOutline size={24} />
                  </Link>
                </li>
                <li>
                  <button data-testid="chat-button" onClick={toggleChat} className="text-black p-3 rounded-full">
                    <IoChatbubblesOutline size={24} />
                  </button>
                  {isChatOpen && (
                    <div className="fixed inset-0 bg-black bg-opacity-50 flex justify-end items-end z-40">
                      <div data-testid="chat-container" className="bg-white w-full max-w-sm md:max-w-lg h-5/6 rounded shadow-lg p-4 relative overflow-hidden mr-10 mb-10">
                        <button data-testid="close-button" onClick={toggleChat} className="absolute top-2 right-2 text-gray-500 hover:text-gray-700">
                          <FaTimes size={24} />
                        </button>
                        <Chat />
                      </div>
                    </div>
                  )}
                </li>
              </>
            )}
          </ul>
        </div>
        <NotificationPane data-testid="notification-pane" open={isModalOpen} onClose={() => setIsModalOpen(false)} />
        <div className="hidden tablet:flex">
          {!isLoggedIn ? (
            <Link to="/login" className="flex gap-2 rounded-full bg-black p-3 px-6">
              <p className="text-white text-sm">Log in</p>
            </Link>
          ) : (
            <button onClick={() => setIsProfileModalOpen(!isProfileModalOpen)} className="text-sm w-[5%]" data-testid="profile-button">
              <RxAvatar className="text-2xl" />
            </button>
          )}
        </div>
        <div className="flex gap-10 items-center">
          <CiSearch className="text-xl flex tablet:hidden" onClick={() => setShowSearch(!showSearch)} />
          {!isLoggedIn ? (
            <Link to="/login" className="flex gap-2 rounded-full bg-black p-3 px-6 tablet:hidden">
              <p className="text-white text-sm">Log in</p>
            </Link>
          ) : (
            <button onClick={() => setIsBurgerShown(!burgerShown)} className="tablet:hidden" data-testid="burger-button">
              {!burgerShown ? <RxHamburgerMenu className="text-3xl" /> : <RxCross2 className="text-3xl" />}
            </button>
          )}
        </div>
  {isLoggedIn && burgerShown && (
  <div className="bg-black flex tablet:hidden z-10 absolute w-[80%] tablet:w-[60%] top-[10%] right-[15%] border border-gray_100 rounded-sm" data-testid="mobile-menu">
    <ul className="w-full p-4 space-y-2">
      <li className="bg-gray rounded-sm">
        <Link to="/" className="text-lg">
          <div className="flex-between justify-center space-x-1 p-2">
            <IoMdHome />
            <p className="text-xs">Home</p>
          </div>
        </Link>
      </li>
      <li className="rounded-sm hover:bg-white">
        <Link
          to=""
          className="text-lg"
          onClick={() => {
            setIsBurgerShown(false);
            setIsModalOpen(true);
          }}
        >
          <div className="flex-between justify-center space-x-1 p-2">
            <Badge badgeContent={unreadCount} color="primary">
              <IoNotificationsOutline
                size={24}
                className="text-gray_100"
              />
            </Badge>
            <p className="text-xs text-gray_100">Notifications</p>
            <NotificationPane
              data-testid="notification-pane"
              open={isModalOpen}
              onClose={() => setIsModalOpen(false)}
            />
          </div>
        </Link>
      </li>
      <li className="rounded-sm hover:bg-white transition-all">
        <Link to="/wishList" className="text-lg">
          <div className="flex-between justify-center space-x-1 p-2">
            <Badge badgeContent={wishListCount} color="primary">
              <CiHeart className="text-gray_100" />
            </Badge>
            <p className="text-xs text-gray_100">My wishlist</p>
          </div>
        </Link>
      </li>
      <li className="rounded-sm hover:bg-white transition-all">
        <Link to="/cart" className="text-lg">
          <div className="flex-between justify-center space-x-1 p-2 ">
            <div className="relative">
              <IoCartOutline className="text-gray_100" />
              {count > 0 && (
                <p className="bg-red-500 p-1 text-1 flex items-center justify-center text-white absolute rounded-full top-[-3px] right-[-3px] text-[8px] h-[13px] w-[13px]">
                  {count}
                </p>
              )}
            </div>
            <p className="text-xs text-gray_100">My cart</p>
          </div>
        </Link>
      </li>
      <li className="rounded-sm hover:bg-white transition-all">
        <Link to="/orders" className="text-lg">
          <div className="flex-between justify-center space-x-1 p-2">
            <IoLocationOutline className="text-gray_100" />
            <p className="text-xs text-gray_100">My Orders</p>
          </div>
        </Link>
      </li>
      <li className="flex items-center justify-center space-x-1 p-2 rounded-sm">
        <button
          data-testid="chat-button"
          onClick={toggleChat}
          className="text-gray flex items-center"
        >
          <Badge color="primary">
            <IoChatbubblesOutline size={17} />
          </Badge>
          <span className="text-xs text-gray_100 ml-1">Chat</span>
        </button>
        {isChatOpen && (
          <div className="fixed inset-0 text-white bg-opacity-50 flex justify-end items-end z-40">
            <div
              data-testid="chat-container"
              className="bg-white w-full max-w-sm md:max-w-lg h-5/6 rounded shadow-lg p-4 relative overflow-hidden mr-10 mb-10"
            >
              <button
                data-testid="close-button"
                onClick={toggleChat}
                className="absolute top-2 right-2 text-black hover:text-gray-700"
              >
                <FaTimes size={24} />
              </button>
              <Chat />
            </div>
          </div>
        )}
      </li>
      <li className="rounded-sm hover:bg-white transition-all">
        <Link
          to="/view-edit-profile"
          className="text-lg"
          onClick={() => setIsBurgerShown(false)}
        >
          <div className="flex-between justify-center space-x-1 p-2">
            <RxAvatar className="text-gray_100" />
            <p className="text-xs text-gray_100">Profile</p>
          </div>
        </Link>
      </li>
      {(authRole === 'seller' || authRole === 'admin') && (
        <li className="rounded-sm hover:bg-white transition-all">
          <Link to={authRole === 'seller' ? '/seller' : '/admin'} className="text-lg">
            <div className="flex-between justify-center space-x-1 p-2">
              <TbDashboard className="text-gray_100" />
              <p className="text-xs text-gray_100">Dashboard</p>
            </div>
          </Link>
        </li>
      )}
      <li className="rounded-sm hover:bg-white transition-all">
        <Link to="/logout" className="text-lg">
          <div className="flex-between justify-center space-x-1 p-2">
            <Logout />
          </div>
        </Link>
      </li>
    </ul>
  </div>
)}
      </nav>
      {isProfileModalOpen && (
  <div className="absolute right-0 bg-gray shadow-lg rounded-lg p-4 m-12 z-50" style={{ minWidth: '250px' }} data-testid="profile-modal">
    <ul>
      <button data-testid="close-button" onClick={() => setIsProfileModalOpen(false)}
        className="absolute top-2 right-2 text-gray-500 hover:text-gray-60">
        <FaTimes size={16} />
      </button>
      <div className="flex flex-col items-center space-y-4">
        {authRole === "admin" || authRole === "seller" ? (
          <Link to={getDashboardLink(authRole)} className="flex items-center space-x-2">
            <TbDashboard size={24} />
            <span>Dashboard</span>
          </Link>
        ) : (
          <Link to="/view-edit-profile" className="flex items-center space-x-2">
            <CiSettings size={20} />
            <span className="text-sm">User settings</span>
          </Link>
        )}
        <Logout />
      </div>
    </ul>
  </div>
)}

      {showSearch && <SearchComponent />}
    </div>
  );
};
 export const getDashboardLink = (authRole:string) => {
    if (authRole === "seller") {
      return "/seller";
    } else if (authRole === "admin") {
      return "/admin";
    } else {
      return "/profile";
    }
  };
export default Navbar