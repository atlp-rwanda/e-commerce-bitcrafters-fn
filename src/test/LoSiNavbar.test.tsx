// Navbar.test.tsx

import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Navbar from "../components/LoSiNavbar";
import Footer from "../components/Footer";
import { BrowserRouter } from "react-router-dom";
import "@testing-library/jest-dom";
import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

describe("Footer component", () => {
  it("renders Footer ", () => {
    const footerTree = renderer
      .create(
        <BrowserRouter>
          <Footer />
        </BrowserRouter>,
      )
      .toJSON();
    expect(footerTree).toMatchSnapshot();
  });
});

const mockStore = configureStore([]);

jest.mock("../assets/images/Bit.Shop.svg", () => "mocked-logo.svg");

describe("Navbar Component", () => {
  let store: any;

  beforeEach(() => {
    store = mockStore({
      auth: { isLoggedIn: true },
      chat: { unreadMessagesCount: 0 },
      cart:  {
        count: 0,
      },
    });
  });

  const renderWithRouter = (component: React.ReactNode) => {
    return render(
      <Provider store={store}>
        <BrowserRouter>{component}</BrowserRouter>
      </Provider>,
    );
  };

  test("renders logo", () => {
    renderWithRouter(<Navbar />);
    const logoElement = screen.getByAltText("Logo");
    expect(logoElement).toBeInTheDocument();
    expect(logoElement).toHaveAttribute("src", "mocked-logo.svg");
  });

  test("renders navigation links on desktop", () => {
    renderWithRouter(<Navbar />);
    const homeLink = screen.getByText("Home");
    const aboutLink = screen.getByText("About");
    const shopLink = screen.getByText("Shop");
    const contactLink = screen.getByText("Contact");

    expect(homeLink).toBeInTheDocument();
    expect(aboutLink).toBeInTheDocument();
    expect(shopLink).toBeInTheDocument();
    expect(contactLink).toBeInTheDocument();
  });
});
