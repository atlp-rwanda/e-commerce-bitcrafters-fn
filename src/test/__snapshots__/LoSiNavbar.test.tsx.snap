// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders Footer  1`] = `
<footer
  className="bg-black w-full "
>
  <div
    className="footer bg-black w-full p-5 px-10 flex flex-row flex-wrap justify-between flex-1 items-start py-12"
  >
    <div
      className="items-start justify-center flex flex-col w-[90%] mb-10 tablet:mb-0 tablet:w-[30%] space-y-4"
    >
      <img
        alt="logo"
        className="r w-24"
        src="mocked-logo.svg"
      />
      <p
        className="text-xs text-gray-400 text-start w-[70%]"
      >
        We have your satisfaction in mind Lorem ipsum dolor sit amet consectetur. In eu aliquet orci ac 
      </p>
      <div
        className="flex-between space-x-2"
      >
        <div
          className="border p-1 border-gray_100 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="LinkedIn"
            >
              <g>
                <path
                  d="M18.44,3.06H5.56a2.507,2.507,0,0,0-2.5,2.5V18.44a2.507,2.507,0,0,0,2.5,2.5H18.44a2.5,2.5,0,0,0,2.5-2.5V5.56A2.5,2.5,0,0,0,18.44,3.06Zm1.5,15.38a1.511,1.511,0,0,1-1.5,1.5H5.56a1.511,1.511,0,0,1-1.5-1.5V5.56a1.511,1.511,0,0,1,1.5-1.5H18.44a1.511,1.511,0,0,1,1.5,1.5Z"
                />
                <g>
                  <path
                    d="M6.376,10.748a1,1,0,1,1,2,0v6.5h0a1,1,0,0,1-2,0Z"
                  />
                  <circle
                    cx="7.376"
                    cy="6.744"
                    r="1"
                  />
                  <path
                    d="M17.62,13.37v3.88a1,1,0,1,1-2,0V13.37a1.615,1.615,0,1,0-3.23,0v3.88a1,1,0,0,1-2,0v-6.5a1.016,1.016,0,0,1,1-1,.94.94,0,0,1,.84.47,3.609,3.609,0,0,1,5.39,3.15Z"
                  />
                </g>
              </g>
            </g>
          </svg>
        </div>
        <div
          className="border p-1 border-gray_100 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 496 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
            />
          </svg>
        </div>
        <div
          className="border p-1 border-gray_100 rounded-full"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              id="Twitter"
            >
              <path
                d="M19.913,5.322a1.034,1.034,0,0,1,.837,1.629L19.708,8.432c-.064,5.086-1.765,8.539-5.056,10.264A10.917,10.917,0,0,1,9.6,19.835a12.233,12.233,0,0,1-6.2-1.524.76.76,0,0,1-.317-.8.768.768,0,0,1,.63-.6,20.6,20.6,0,0,0,3.745-.886C2,13.5,3.19,7.824,3.71,6.081a1.028,1.028,0,0,1,1.729-.422,9.931,9.931,0,0,0,5.995,2.95A4.188,4.188,0,0,1,12.725,5.3a4.125,4.125,0,0,1,5.7.02ZM4.521,17.794c1.862.872,6.226,1.819,9.667.016,2.955-1.549,4.476-4.732,4.521-9.461a.771.771,0,0,1,.142-.436l1.081-1.538-.041-.053c-.518-.007-1.029-.014-1.55,0a.835.835,0,0,1-.547-.221,3.13,3.13,0,0,0-4.383-.072,3.174,3.174,0,0,0-.935,2.87.646.646,0,0,1-.154.545.591.591,0,0,1-.516.205A10.924,10.924,0,0,1,4.722,6.354c-.67,2.078-1.52,7.094,3.869,9.065a.632.632,0,0,1,.416.538.625.625,0,0,1-.3.6A13.178,13.178,0,0,1,4.521,17.794ZM11.875,8.65h0Zm7.793-.161,0,0Z"
              />
            </g>
          </svg>
        </div>
      </div>
    </div>
    <div
      className="items-start justify-center flex flex-col w-[40%] bigphone:w-[12%] space-y-2 mb-10 phone:mb-0"
    >
      <p
        className="text-gray-200 font-semibold text-sm text-white"
      >
        Account
      </p>
      <p
        className=" text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          My Account
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Login / Register
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Cart
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Wishlist
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Shop
        </a>
      </p>
    </div>
    <div
      className="items-start justify-center flex flex-col w-[40%] bigphone:w-[12%] space-y-2 mb-10 phone:mb-0"
    >
      <p
        className="text-white font-semibold text-sm"
      >
        Information
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/FAQ"
          onClick={[Function]}
        >
          FAQ
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/about"
          onClick={[Function]}
        >
          About Us
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Resources
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Terms
        </a>
      </p>
      <p
        className="text-xs  text-start hover:white transition-all"
      >
        <a
          href="/privacyPolicy"
          onClick={[Function]}
        >
          Privacy Policy
        </a>
      </p>
    </div>
    <div
      className="items-start justify-center flex flex-col w-[80%] bigphone:w-[25%]  mt-4 bigphone:mt-1 space-y-2 mb-10 phone:mb-0"
    >
      <p
        className="text-white font-semibold text-sm"
      >
        Contact Us
      </p>
      <div
        className="flex-between items-start space-x-2"
      >
        <svg
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "white",
            }
          }
          viewBox="0 0 448 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M380.9 97.1C339 55.1 283.2 32 223.9 32c-122.4 0-222 99.6-222 222 0 39.1 10.2 77.3 29.6 111L0 480l117.7-30.9c32.4 17.7 68.9 27 106.1 27h.1c122.3 0 224.1-99.6 224.1-222 0-59.3-25.2-115-67.1-157zm-157 341.6c-33.2 0-65.7-8.9-94-25.7l-6.7-4-69.8 18.3L72 359.2l-4.4-7c-18.5-29.4-28.2-63.3-28.2-98.2 0-101.7 82.8-184.5 184.6-184.5 49.3 0 95.6 19.2 130.4 54.1 34.8 34.9 56.2 81.2 56.1 130.5 0 101.8-84.9 184.6-186.6 184.6zm101.2-138.2c-5.5-2.8-32.8-16.2-37.9-18-5.1-1.9-8.8-2.8-12.5 2.8-3.7 5.6-14.3 18-17.6 21.8-3.2 3.7-6.5 4.2-12 1.4-32.6-16.3-54-29.1-75.5-66-5.7-9.8 5.7-9.1 16.3-30.3 1.8-3.7.9-6.9-.5-9.7-1.4-2.8-12.5-30.1-17.1-41.2-4.5-10.8-9.1-9.3-12.5-9.5-3.2-.2-6.9-.2-10.6-.2-3.7 0-9.7 1.4-14.8 6.9-5.1 5.6-19.4 19-19.4 46.3 0 27.3 19.9 53.7 22.6 57.4 2.8 3.7 39.1 59.7 94.8 83.8 35.2 15.2 49 16.5 66.6 13.9 10.7-1.6 32.8-13.4 37.4-26.4 4.6-13 4.6-24.1 3.2-26.4-1.3-2.5-5-3.9-10.5-6.6z"
          />
        </svg>
        <div>
          <p
            className="text-xs text-gray-200 text-start text-white"
          >
            Whatsapp
          </p>
          <p
            className="text-xs text-gray-400 text-start hover:text-gray-200 transition-all "
          >
            <a
              href="/"
              onClick={[Function]}
            >
              +250 788 888 888
            </a>
          </p>
        </div>
      </div>
      <div
        className="flex-between items-start space-x-2"
      >
        <svg
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "white",
            }
          }
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M164.9 24.6c-7.7-18.6-28-28.5-47.4-23.2l-88 24C12.1 30.2 0 46 0 64C0 311.4 200.6 512 448 512c18 0 33.8-12.1 38.6-29.5l24-88c5.3-19.4-4.6-39.7-23.2-47.4l-96-40c-16.3-6.8-35.2-2.1-46.3 11.6L304.7 368C234.3 334.7 177.3 277.7 144 207.3L193.3 167c13.7-11.2 18.4-30 11.6-46.3l-40-96z"
          />
        </svg>
        <div>
          <p
            className="text-xs text-gray-200 text-start  text-white"
          >
            Call Us
          </p>
          <p
            className="text-xs text-gray-400 text-start hover:text-gray-200 transition-all "
          >
            <a
              href="tel:+250 788 888 888"
            >
              +250 788 888 888
            </a>
          </p>
        </div>
      </div>
      <div
        className="flex-between items-start space-x-2"
      >
        <svg
          color="white"
          fill="currentColor"
          height={20}
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": "white",
            }
          }
          viewBox="0 0 24 24"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
          />
        </svg>
        <div>
          <p
            className="text-xs text-gray-200 text-start  text-white"
          >
            Email Us
          </p>
          <p
            className="text-[10px] tablet:text-xs text-gray-400 text-start hover:text-gray-200 transition-all "
          >
            <a
              href="/bitcrafter.andela@gmail.com"
              onClick={[Function]}
            >
              bitcrafter.andela@gmail.com
            </a>
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="p-6 flex items-center w-[80%] m-auto justify-center border-t-[1px] border-gray_100"
  >
    <p
      className="text-xs text-gray-400"
    >
      © 
      2024
       All rights reserved - Bitcrafters Andela Team
    </p>
  </div>
</footer>
`;
